hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -249,6 +249,19 @@
           + "record size estimate compute dynamically based on commit metadata. "
           + " This is critical in computing the insert parallelism and bin-packing inserts into small files.");
 
+  public static final ConfigProperty<String> MAX_ARCHIVE_FILES_TO_KEEP_PROP = ConfigProperty
+      .key("hoodie.max.archive.files")
+      .defaultValue("10")
+      .withDocumentation("The numbers of kept archive files under archived.");
+
+  public static final ConfigProperty<String> AUTO_TRIM_ARCHIVE_FILES_DROP = ConfigProperty
+      .key("hoodie.auto.trim.archive.files")
+      .defaultValue("false")
+      .withDocumentation("When enabled, Hoodie will keep the most recent " + MAX_ARCHIVE_FILES_TO_KEEP_PROP.key()
+          + " archive files and delete older one which lose part of archived instants information.");
```

Comment:
Should these configs live in `HoodieWriteConfig` instead of `HoodieCompactionConfig`?

Reply:
Emmm, because all the archive related configs such as `hoodie.archive.automatic`, `hoodie.commits.archival.batch` and `hoodie.keep.min.commits`, etc are all lived in `HoodieCompactionConfig `, maybe it's better to be the same :)

Reply:
Got it.  Ideally, the archive configs should not be in `HoodieCompactionConfig`.  Let's keep it as is for now and clean this up in a follow-up PR.